lisp/ess-r-flymake.el

```diff
@@ -76,28 +76,26 @@ each element is passed as argument to `lintr::linters_with_defaults'."
 (defvar-local ess-r--flymake-proc nil)
 
 (defvar-local ess-r--lintr-file nil
-  "Location of the .lintr file for this buffer.")
+  "Location of the .lintr config file for this buffer.")
 
 (defvar ess-r--flymake-def-linter
   (replace-regexp-in-string
    "[\n\t ]+" " "
    "esslint <- function(str, ...) {
-    if (!suppressWarnings(require(lintr, quietly=T))) {
+    if (!suppressWarnings(requireNamespace('lintr', quietly=TRUE))) {
         cat('@@error: @@`lintr` package not installed')
+    } else if (packageVersion('lintr') <= '3.0.0') {
+        cat('@@error: @@Need `lintr` version > v3.0.0')
     } else {
-        if (packageVersion('lintr') <= '3.0.0') {
-            cat('@@error: @@Need `lintr` version > v3.0.0')
-        } else {
-            tryCatch(lintr::lint(commandArgs(TRUE), ...),
-                    error = function(e) {
-                       cat('@@warning: @@', conditionMessage(e))
-                    })
-        }
+        tryCatch(lintr::lint(text=commandArgs(TRUE), ..., parse_settings=TRUE),
```

Comment:
While at it I think this should be changed to `text = str`, so the function argument is actually used.